dev/AppAttach/Extension/AppAttachExtension/Providers/IDEResourceProvider.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation and Contributors.
+// Licensed under the MIT License.
+
+using AppAttachExtension.Models;
+using EnvDTE80;
+using Microsoft.VisualStudio.Setup.Configuration;
+using Microsoft.VisualStudio.Shell.Interop;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace AppAttachExtension.Providers
+{
+    public class IDEResourceProvider
+    {
+        private readonly IDEResouceModel _model;
+
+        public IDEResourceProvider()
+        {
+            _model = new IDEResouceModel();
+        }
+
+        public async Task PopulateIDEModelAsync()
+        {
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+            var _dte2 = await ServiceProvider.GetGlobalServiceAsync(typeof(SDTE)) as DTE2;
+            var selectedItem = _dte2.SelectedItems.Item(1);
+            _model.ProjectPath = selectedItem.Project.FullName;
+            var vsVersion = _dte2.Version.Split('.')[0];
+            var vsEdition = _dte2.Edition;
+            var filePath = "";
+            var query = new SetupConfiguration();
+            var query2 = (ISetupConfiguration2)query;
+            var e = query2.EnumAllInstances();
+
+            var helper = (ISetupHelper)query;
+            List<string> list = new List<string>();
+            int fetched;
+            var instances = new ISetupInstance[1];
+            do
+            {
+                e.Next(1, instances, out fetched);
+                if (fetched > 0)
+                    list.Add(instances[0].GetInstallationPath());
+            } while (fetched > 0);
+            string version = vsVersion == "17" ? "2022" : "2019";

```

Comment:
Does the AppAttachExtension need to support VS 2019?

@bpulliam @MikeHillberg @Scottj1s Where's WinAppSDK's policy documented regarding supported tools? Version 1.2's release notes has [this comment](https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/stable-channel#version-12-1)

>Visual Studio 2019 and .NET 5 is no longer supported for building C# apps (see [Windows App SDK 1.2 moving to C# WinRT 2.0](https://github.com/microsoft/WindowsAppSDK/discussions/2879)). You will need Visual Studio 2022...

and there's no VSIX for 2019 for new WinAppSDK releases. It's unclear how the AppAttach VS Extension fits. Do we need a policy change? Is this an exception to the rule - if someone in WinAppSDK wants to support older tools (or Windows releases), can they? Do they need approval to not meet the standard policies (in this case, more expansive, vs the usual less expansive cases)?

Something we should follow up offline?

Reply:
> Does the AppAttachExtension need to support VS 2019?
> 
> @bpulliam @MikeHillberg @Scottj1s Where's WinAppSDK's policy documented regarding supported tools? Version 1.2's release notes has [this comment](https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/stable-channel#version-12-1)
> 
> > Visual Studio 2019 and .NET 5 is no longer supported for building C# apps (see [Windows App SDK 1.2 moving to C# WinRT 2.0](https://github.com/microsoft/WindowsAppSDK/discussions/2879)). You will need Visual Studio 2022...
> 
> and there's no VSIX for 2019 for new WinAppSDK releases. It's unclear how the AppAttach VS Extension fits. Do we need a policy change? Is this an exception to the rule - if someone in WinAppSDK wants to support older tools (or Windows releases), can they? Do they need approval to not meet the standard policies (in this case, more expansive, vs the usual less expansive cases)?
> 
> Something we should follow up offline?

Sure, Will go through all the comments, and will setup time for offline discussion.